src/Member/MemberUtil.php

```diff
@@ -172,4 +174,49 @@ public function findOrCreate(string $email)
 
         return $member;
     }
+
+    /**
+     * @return \Contao\Model[]|null
+     */
+    public function getActiveGroups(int $memberId, array $options = [])
+    {
+        if (!$memberId) {
+            return null;
+        }
+
+        $modelUtil = System::getContainer()->get(ModelUtil::class);
+
+        if (null === ($memberModel = $modelUtil->findModelInstanceByIdOrAlias('tl_member', $memberId, $options))) {
+            return null;
+        }
+
+        $groups = StringUtil::deserialize($memberModel->groups, true);
+
+        $columns = ['tl_member_group.id IN('.implode(',', array_map('\intval', $groups)).')'];
+
+        $modelUtil->addPublishedCheckToModelArrays('tl_member_group', 'disable', 'start', 'stop', $columns, ['invertPublishedField' => true]);
+
+        if (null === ($groupModelCollection = $modelUtil->findModelInstancesBy('tl_member_group', $columns, []))) {
+            return null;
+        }
+
+        return $groupModelCollection->getModels();
+    }
+
+    public function hasActiveGroup(int $memberId, int $group): bool
+    {
+        $activeGroups = $this->getActiveGroups($memberId);
+
+        if (empty($activeGroups)) {
```

Comment:
Unnecessary if the method above always returns an array. Just throw it into the for loop and there you go.